### ISSUE TITLE
utf8_decode() expects parameter 1 to be string, array given 

### ISSUE THREAD

**Issue by yiqing-95** [MollsReis/junior#21]
in Response class 

```
 // create a new json rpc response object
13     public function __construct($result, $id = null, $error_code = null, $error_message = null)
14     {
15         $this->result = utf8_decode($result);
```

don ' t know why the error occurs , but seems that the $result is an array  

in Client class

```

        // try to decode json
        $json_response = $this->decodeJSON($response);
```

may be the result can be an array from above code .

if so  the utf8_decode  should recursively decode the key and val .

**Comment by MollsReis** [comment]
I'm not sure I understand your issue. Can you give me an example of this error occurring during use? Can you give me any details of the messages you are sending/receiving with Junior and what version of PHP you are using?

**Comment by yiqing-95** [comment]
i am the client and server end is a dot net jsonRpc2.0 implement . firstly i use another php jsonRpc implements it works well   but it seems too simple for server side  so i switch to this one . 

when  i debug the message transmitting  i use a http proxy  tool (tcpmon)[http://ws.apache.org/tcpmon/] and  i am sure that  the message is json format 

i am using php version is 5.3.8    

```
15         $this->result = utf8_decode($result);
```

when i change above to :           $this->result = is_string($result)? utf8_decode($result) : $result ;
it works but i still think  if it should decode recursively ?

**Comment by MollsReis** [comment]
I think I understand the issue now. This was coded assuming that the result field in a response would be a simple string. I'm guessing you response looks something more like: { "jsonrpc": "2.0", "result": [1, 2, 3], "id": 1 }

I will write a test to handle this case and update the code. Thanks for contributing! :)

**Comment by MollsReis** [comment]
Your issue is solved in v1.1.3. You can get the latest version from composer or clone it from github. Thanks again!

**Comment by yiqing-95** [comment]
yes  the format  you guessed is actually  right !  thanks for your great works  too !
